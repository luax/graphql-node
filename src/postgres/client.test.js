const { client, sql } = require("./index");

describe("client", () => {
  beforeAll(() => {
    client.initialize();
  });

  afterAll(async () => {
    await client.end();
  });

  test("returns now", async () => {
    const res = await client.query(sql`SELECT NOW() as now`);
    expect(res).toBeTruthy();
  });

  test("returns an error", async () => {
    expect(
      client.query(sql`SELECT NOW as now`),
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  test("handles sql injection", async () => {
    const userInput = "; DROP TABLE authors; as foo";
    const res = await client.query(sql`SELECT ${userInput} as test`);
    expect(res[0].test).toEqual(userInput);
    expect(
      client.query(sql`SELECT ${sql.identifier([userInput])} as test`),
    ).rejects.toThrow(`column "${userInput}" does not exist`);
  });
});
