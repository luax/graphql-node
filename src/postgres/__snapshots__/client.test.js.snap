// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client handles sql injection 1`] = `"column \\"now\\" does not exist"`;
